content/client-lib-development-guide/features.textile

```diff
@@ -84,6 +84,11 @@ h3(#restclient). RestClient
 * @(RSC8)@ Supports two protocols:
 ** @(RSC8a)@ "MessagePack":http://msgpack.org/ binary protocol (this is the default for environments having a suitable level or support for binary data)
 ** @(RSC8b)@ JSON text protocol (used when @useBinaryProtocol@ option is false)
+** @(RSC8c)@ The library sets the @Accept@ and @Content-Type@ type headers, to reflect whether the client is configured to use a binary or JSON based protocol, to @application/x-msgpack@ or @application/json@ respectively.
+** @(RSC8d)@ Irrespective of the specified @Accept@ header, the library should attempt where possible to process any response body based on the content type indicated in a @Content-Type@ header.
```

Comment:
Again, I think this could be more specific...

```suggestion
** @(RSC8d)@ Irrespective of the @Accept@ header set by the library (see "RSC8c":#RSC8c), it should attempt where possible to process any response body based on the received @Content-Type@ header.
```

However, now that I even read that back to myself, I'm failing to find something specific that I could implement in code around this. What does this mean in practice? Are we saying that the client library needs to have multiple parsers handling different possibly received content types? Or is it meant to mangle say `text/html` to `application/json` because that's what it specified in `Accept`? **I think this is too ambiguous.**

Reply:
If the client says it `Accept`s JSON, but the server responds with msgpack, it should parse it, and vice versa, if the library already contains support for the alternate format. It should not add support for parsing anything (eg html) just in order to be able to process erroneously-formatted responses.

Reply:
improved https://github.com/ably/docs/pull/888/commits/bfcb292156fecf01102086c64cd951f7b018c8b5